File: packages/reown_yttrium/ios/Classes/ReownYttriumPlugin.swift
import Flutter
import UIKit

public class ReownYttriumPlugin: NSObject, FlutterPlugin {
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "reown_yttrium", binaryMessenger: registrar.messenger())
        let instance = ReownYttriumPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "erc6492_init":
            EIP6492Verifier.initialize(call.arguments ?? {}, result: result)
        case "erc6492_verify":
            EIP6492Verifier.verifySignature(call.arguments ?? {}, result: result)
        case "erc6492_dispose":
            EIP6492Verifier.dispose(call.arguments ?? {}, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
    

File: packages/reown_yttrium/android/src/main/kotlin/com/reown/reown_yttrium/ReownYttriumPlugin.kt
package com.reown.reown_yttrium

import android.content.Context

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** ReownYttriumPlugin */
class ReownYttriumPlugin: FlutterPlugin, MethodCallHandler {
  private lateinit var channel : MethodChannel
  private lateinit var applicationContext: Context // ✅ Store application context

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    applicationContext = flutterPluginBinding.applicationContext // ✅ Get application context
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "reown_yttrium")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    when (call.method) {
      "erc6492_init" -> EIP6492Verifier.init(call.arguments, result)
      "erc6492_verify" -> EIP6492Verifier.verifySignature(call.arguments, result)
      "erc6492_dispose" -> EIP6492Verifier.dispose(call.arguments, result)
      "wcp_init" -> WalletConnectPay.initialize(call.arguments, result)
      "wcp_createPayment" -> WalletConnectPay.createPayment(call.arguments, result)
      else -> result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
